### PR TITLE
[Confluence Documentation] Update message AddAssetQuantity and SubtractAssetQuantity with extra optional parameter

### PR DESCRIPTION
**Pull request by dominious1**
Enhance functionality in the C++ Iroha base core code by introducing an optional argument, "title," to the AddAssetQuantity and SubtractAssetQuantity commands within the commands.proto bufs. Ensure seamless integration and functionality of this optional parameter, and optimize associated Gtests to accommodate these modifications. Additionally, enhance the Python client, contained in a separate repository, to effectively handle these changes. This refinement contributes to a more robust and adaptable system, aligning with the evolving requirements of the codebase.

Link to C++ library:
https://github.com/hyperledger/iroha/pull/4003/files

Link to Python client library which use protobuf GRPC shared code base:
https://github.com/hyperledger/iroha-python/

This is part of internship project:
https://wiki.hyperledger.org/display/INTERN/Iroha+1%3A+extend+queries+with+optional+arguments

Link to PR in iroha-python compatible with those changes:
hyperledger/iroha-python#157

Base documentation:
https://iroha.readthedocs.io/en/develop/develop/api/commands.html?highlight=AddAssetQuantity#id80

### CODE REVIEW

File: docs/source/develop/api/commands.rst [hunk 1]
```diff
@@ -38,6 +38,7 @@ Structure
 
     "Asset ID", "id of the asset", "<asset_name>#<domain_id>", "usd#morgan"
     "Amount", "positive amount of the asset to add", "> 0", "200.02"
+    "Title", "title of the tranzaction", "title", "title"
```
Comment (andprogrammer) [inline docs/source/develop/api/commands.rst]: Typo tranzaction -> transaction

File: docs/source/develop/api/commands.rst [hunk 2]
```diff
@@ -810,6 +811,7 @@ Structure
 
     "Asset ID", "id of the asset", "<asset_name>#<domain_id>", "usd#morgan"
     "Amount", "positive amount of the asset to subtract", "> 0", "200"
+    "Title", "title of the tranzaction", "title", "title"
```
Comment (andprogrammer) [inline docs/source/develop/api/commands.rst]: Typo tranzaction -> transaction

File: docs/source/develop/api/commands.rst [hunk 3]
```diff
@@ -21,6 +21,7 @@ Schema
     message AddAssetQuantity {
         string asset_id = 1;
         string amount = 2;
+        optional string title = 3;
     }
```
Comment (baziorek) [inline docs/source/develop/api/commands.rst:25]: I suggest to use: `.. literalinclude:: ../../../../shared_model/schema/commands.proto`, similar usage You have in file `queries.rst`. But this is not what You must change - this is up to You.

**Review by baziorek — DISMISSED** [review]
It looks OK for me:)

File: docs/source/develop/api/commands.rst (hunk 1 above)
Comment (dominious1) [reply docs/source/develop/api/commands.rst]: done

File: docs/source/develop/api/commands.rst (hunk 2 above)
Comment (dominious1) [reply docs/source/develop/api/commands.rst]: done

File: docs/source/develop/api/commands.rst [hunk 4]
```diff
@@ -37,6 +38,7 @@ Structure
 
     "Asset ID", "id of the asset", "<asset_name>#<domain_id>", "usd#morgan"
     "Amount", "positive amount of the asset to add", "> 0", "200.02"
+    "Title", "title of the transaction", "title", "title"
```
Comment (Mingela) [inline docs/source/develop/api/commands.rst:41]: `transaction` may be confusing here, that's applicable to the command, right?

Comment (6r1d) [reply docs/source/develop/api/commands.rst:41]: @dominious1 any comments?

**Review by 6r1d — CHANGES_REQUESTED** [review]
Please address Bogdan's comment, LGTM otherwise

**Comment by baziorek** [comment]
@dominious1 The main PR https://github.com/hyperledger/iroha/pull/4003 was merged. So please update this PR https://github.com/hyperledger/iroha/pull/4032 to have the same names

File: docs/source/develop/api/commands.rst [hunk 5]
```diff
@@ -37,6 +38,7 @@ Structure
 
     "Asset ID", "id of the asset", "<asset_name>#<domain_id>", "usd#morgan"
     "Amount", "positive amount of the asset to add", "> 0", "200.02"
+    "Description", "description of the transaction", "description", "description"
```
Comment (baziorek) [inline docs/source/develop/api/commands.rst:41]: Please correct those cells:
```suggestion
    "Description", "description of the transaction", "Max length of description (set in genesis block, by default is 100*1024)", "Mint assets"
```

File: docs/source/develop/api/commands.rst [hunk 6]
```diff
@@ -808,6 +811,7 @@ Structure
 
     "Asset ID", "id of the asset", "<asset_name>#<domain_id>", "usd#morgan"
     "Amount", "positive amount of the asset to subtract", "> 0", "200"
+    "Description", "description of the transaction", "description", "description"
```
Comment (baziorek) [inline docs/source/develop/api/commands.rst:814]: similar as above

**Review by baziorek — DISMISSED** [review]

File: docs/source/develop/api/commands.rst (hunk 6 above)
Comment (baziorek) [reply docs/source/develop/api/commands.rst:814]:
```suggestion
    "Description", "description of the transaction", "Max length of description (set in genesis block, by default is 100*1024)", "Burn assets"
```

**Review by baziorek — DISMISSED** [review]
LGTM

File: docs/source/develop/api/commands.rst (hunk 4 above)
Comment (baziorek) [reply docs/source/develop/api/commands.rst]: it was corrected in:  https://github.com/hyperledger/iroha/pull/4032/commits/59aecd8908786a699e650dd8b368401cb4222ebb

**Review by 6r1d — APPROVED** [review]
LGTM

**Review by baziorek — DISMISSED** [review]
LGTM:).

Previous approves disappeared because I needed to change destination branch. Branch `develop` does not contain changes required for build (dependencies are broken).

Previous approves:
https://github.com/hyperledger/iroha/pull/4032#pullrequestreview-1838202932
https://github.com/hyperledger/iroha/pull/4032#pullrequestreview-1837444380

**Review by baziorek — APPROVED** [review]
after fixing  DCO it still LGTM:D

**Comment by baziorek** [comment]
@6r1d Yesterday I had meeting with @dominious1 and @andprogrammer. We finally fixed DCO. Comments from @Mingela (https://github.com/hyperledger/iroha/pull/4032#discussion_r1401856085) were fixed. DCO and all other checks are green. Previous PR to the code (https://github.com/hyperledger/iroha/pull/4003) is merged.
IMO it is ready to merge.

**Review by 6r1d — APPROVED** [review]
LGTM

**Pull request merged**